src/Eshopworld.WorkerProcess/Stores/CosmosDbLeaseStore.cs

```diff
@@ -45,7 +45,7 @@ await _documentClient.CreateDatabaseIfNotExistsAsync(new Database {Id = _options
                 Id = _options.Value.Collection
             };
 
-            collectionDefinition.PartitionKey.Paths.Add("/instanceId");
+            collectionDefinition.PartitionKey.Paths.Add("/leaseType");
```

Comment:
I may have a bunch of read up here before I have context so do bare this in mind with my questions. Going from instance id to lease type sounds like a reduction in entrpy and thus may cause higher levels of hot partitions when underload?

Reply:
The amount of data in the collection is minuscule, as is the read write rates. Partitioning is not really required. Nothing will ever be hot unless something is really broken, lease allocation in the millisecond range.

Reply:
Thanks for this guys. I'd agree that partitioning isn't much of a concern and workerprocess only creates one document per leaseType. We would only have at most a handful of workertypes in a cosmos domain as well.

Removing partition key is an option but as unpartioned collections are now legacy, I would prefer to avoid if possible. 

I had though of another id in hosting application (guid or otherwise) that would be managed within hosting application. In end this just seems like extra work and not seeing benefit of use of leasetype as parititon key. If anyone has any ideas around that approach, I would be happy to consider however.